packages/ee/embed-sdk/src/index.ts

```diff
@@ -45,119 +55,211 @@ export const jwtTokenQueryParamName = "jwtToken"
 class ActivepiecesEmbedded {
   _prefix = '';
   _initialRoute = '';
+  _instanceUrl = '';
   _hideSidebar = false;
   _disableNavigationInBuilder = true;
-  iframeParentOrigin = window.location.origin;
   handleVendorNavigation?: (data: { route: string }) => void;
   handleClientNavigation?: (data: { route: string }) => void;
   parentOrigin = window.location.origin;
+  private createIframe({src,style}:{src:string,style:string})
+  {
+    const iframe = document.createElement('iframe');
+    iframe.src = src;
+    iframe.style.cssText = style;
+    return iframe;
+  }
+  private connectoToEmbed ({instanceUrl,jwtToken,iframeContainer,iframeStyling,client,name}: {
+    instanceUrl: string,
+    jwtToken: string,
+    iframeStyling: string | undefined,
+    iframeContainer: Element,
+    client: ActivepiecesEmbedded,
+    name:string
+  }
+  ): IframeWithWindow {
+    const iframe = this.createIframe({src:`${instanceUrl}/embed?${jwtTokenQueryParamName}=${jwtToken}`,style:iframeStyling || ''});
+    iframeContainer.appendChild(iframe);
+    if (!this.doesFrameHaveWindow(iframe)) {
+      const error = 'Activepieces: iframe window not accessible';
+      console.error(error);
+      throw new Error(error);
+    }
+    const iframeWindow = iframe.contentWindow;
+      window.addEventListener(
+        'message',
+        function (event: MessageEvent<ActivepiecesClientEvent>) {
+     
+          if (event.source === iframeWindow) {
+            switch (event.data.type) {
+              case ActivepiecesClientEventName.CLIENT_INIT: {
+                const apEvent: ActivepiecesVendorInit = {
+                  type: ActivepiecesVendorEventName.VENDOR_INIT,
+                  data: {
+                    prefix: client._prefix,
+                    initialRoute:  client._initialRoute,
+                    hideSidebar: client._hideSidebar,
+                    disableNavigationInBuilder: client._disableNavigationInBuilder,
+                  },
+                };
+                iframeWindow.postMessage(apEvent, '*');
+                break;
+              }
+            }
+          }
+        }
+      );
+     return iframe;
+  };
+  _connectionsIframe: HTMLIFrameElement | null = null;
   configure({
     prefix,
     hideSidebar,
-    disableNavigationInBuilder
+    disableNavigationInBuilder,
+    builderIframeContainerSelector,
```

Comment:
`builderContainerId` should be enough here, there is no need for them to know about iframe or the selector